### ISSUE TITLE
Cannot use dot in arg, yargs-parser dot-notation option is enabled by default

### ISSUE THREAD

**Issue by unlight** [mozilla/node-convict#240]
https://github.com/yargs/yargs-parser#dot-notation

Part of schema:
```
        database: {
            default: ':memory:',
            arg: 'database.name',
        },
```
CLI:
```
// Fails
... --database.name=db
```
Because `argv` is object = `{database: {name: 'db'} }`
`if (argv[argStr])` is always false.

Fix or mention such behavior in documentation.

**Comment by PatrickShaw** [comment]
Added two PRs that fix this issue.
The first approach is faster - Disables yarg-parser's dot-notation parsing option.
#269 

The second approach is slightly slower - Splits node-convict's `argStr` by `"."` and traverses the object returned from yarg-parser.
#270

**Comment by PatrickShaw** [comment]
#269

**Comment by madarche** [comment]
Fixed by #269.
